nixos/modules/security/pam.nix

```diff
@@ -218,7 +218,7 @@ let
       # Samba stuff to the Samba module.  This requires that the PAM
       # module provides the right hooks.
```

Comment:
Here, I would add something like the line below. That would ensure that you get a 'compile'-time error. if you set 'unixAuth' manually somewhere else.

# ECryptFS and PamMount require unixAuth.
unixAuth = mkIf (config.security.pam.enableEcryptfs || cfg.pamMount) true;